### PR TITLE
Change LandmarkGroup dict to OrderedDict

### PR DESCRIPTION
**Pull request by grigorisg9gr**
This is a tiny practical change. 

It is particularly  valuable for detection/tracking when there are multiple bounding boxes/landmarks returned. Usually, those are sorted based on confidence, so it makes sense to have them sorted based on the detectors' confidence. 
For instance, calling  `im.landmarks.group_labels[0]` should return the most confident detection. 

While this can be done when calling it in the user code, I believe it is a sensible change in the group_labels.

### CODE REVIEW

**Comment by jabooth** [comment]
@grigorisg9gr I could see the use case for this, but if we include it can you also update the documentation to state that the labels are always returned sorted?

**Comment by grigorisg9gr** [comment]
Hope this is the documentation you had in mind @jabooth . Thanks for the remark.

**Comment by patricksnape** [comment]
Do we want to sort these keys directly - or would we actually prefer an `OrderedDict`?

**Comment by nontas** [comment]
I think an `OrderedDict` makes more sense.

**Comment by patricksnape** [comment]
OK, only note about that is we would also need to update ``__reduce__`` as in `PCAModel` in order to ensure previously saved landmarks come back with `OrderedDict`.

**Comment by jabooth** [comment]
I know it's early days and it's a ways away but there is this on the horizon... https://twitter.com/raymondh/status/773978885092323328

**Comment by patricksnape** [comment]
Yes but that is an implementation detail! Not to be relied upon!

File: menpo/landmark/base.py [hunk 1]
```diff
@@ -219,6 +219,13 @@ def __delitem__(self, group):
     def __len__(self):
         return len(self._landmark_groups)
 
+    def __setstate__(self, state):
+        # consistency with older versions imported.
+        if ~isinstance(state['_landmark_groups'], OrderedDict):
```
Comment (patricksnape) [inline menpo/landmark/base.py]: I think this should be the `not` keyword rather than binary not (`~`).

File: menpo/landmark/base.py [hunk 2]
```diff
@@ -240,7 +247,7 @@ def has_landmarks(self):
     @property
     def group_labels(self):
         """
-        All the labels for the landmark set.
+        All the labels for the landmark set sorted alphabetically.
```
Comment (patricksnape) [inline menpo/landmark/base.py]: Not sorted alphabetically - sorted by insertion order.

**Comment by patricksnape** [comment]
Ok this is finally ready to be reviewed by @jabooth

**Pull request merged**